File: src/models/lift_stats.rs
use std::collections::BTreeMap;

use super::LiftExecution;
use crate::weight::Weight;

/// Summary statistics for a lift used by the query interface.
#[derive(Debug)]
pub struct LiftStats {
    /// Most recent execution of the lift, if any.
    pub last: Option<LiftExecution>,
    /// Heaviest weight performed for each repetition count.
    pub best_by_reps: BTreeMap<i32, Weight>,
}

File: java/src/main/java/org/lift/trax/LiftStats.java
package org.lift.trax;

import java.util.Map;

public class LiftStats {
    public LiftExecution last;
    public Map<Integer, Double> bestByReps;

    public LiftStats(LiftExecution last, Map<Integer, Double> bestByReps) {
        this.last = last;
        this.bestByReps = bestByReps;
    }
}
